Clarify all default components for C++ workload (#33870)
dotnet/docs: docs/core/deploying/native-aot/index.md

The following prerequisites need to be installed before publishing .NET projects with native AOT.

> On Windows, install Windows SDK and [Visual Studio 2022](https://visualstudio.microsoft.com/vs/), including Desktop development with C++ workload.

Reviewer: ```suggestion
On Windows, install [Visual Studio 2022](https://visualstudio.microsoft.com/vs/), including Desktop development with C++ workload with all default components.
```
We should keep the simple instructions first. They take care of Windows SDK too. People do not need to install Windows SDK explicitly if they just follow the simple instructions.

If you think we need advanced instructions, they should be second and clearly marked optional and not something non-advanced users need to worry about.

Author: Makes sense; it clearly reads "default selection is the minimum requirement".

When I selected `Desktop development with C++`, the default selection had the latest Windows 11 SDK on Windows 10, but it was still enough for publishing NativeAOT apps.

![image](https://user-images.githubusercontent.com/3840695/217057691-06d4c794-5907-45cf-832a-0cf0eca6789d.png)
